### PR TITLE
removed from models import* from bodhi.server.models.__init__

### PR DESCRIPTION
**Pull request by ankit01ojha**
removed the pacjkage bodhi.server.models and replaced it with bodhi.server.models.py.

### CODE REVIEW

**Comment by ankit01ojha** [comment]
Sorry for four commits, I'am new to github :(

File: bodhi/server/models.py [hunk 1]
```diff
@@ -1,3 +1,145 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+from bodhi.server.models import models
```
Comment (trishnaguha) [inline bodhi/server/models.py]: We don't really need this inside  `models.py`

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Comment by trishnaguha** [comment]
Could you move all the import statements to the top? https://github.com/ankit01ojha/bodhi/blob/develop/bodhi/server/models.py#L159-L205

**Comment by trishnaguha** [comment]
We don't need to import from `enum` anymore since it is inside `models.py` now https://github.com/ankit01ojha/bodhi/blob/develop/bodhi/server/models.py#L196

**Comment by trishnaguha** [comment]
Please fix the branch conflict and rebase.

**Comment by trishnaguha** [comment]
@ankit01ojha Did you run the test suite? If not can you please run the test suite? Because I think some of the tests are going to fail because of the import.

**Comment by ankit01ojha** [comment]
Thanks @trishnaguha  i'll correct them.

File: bodhi/server/models.py [hunk 2]
```diff
@@ -62,6 +46,167 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# http://techspot.zzzeek.org/2011/01/14/the-enum-recipe
+
+import re
+
+from sqlalchemy.types import SchemaType, TypeDecorator, Enum
```
Comment (trishnaguha) [inline bodhi/server/models.py]: The idea is to keep all the import statements together. https://github.com/ankit01ojha/bodhi/blob/b7fdff2e6f66d8a487848f8424189bec798a55c9/bodhi/server/models.py#L1-L83
Please check the import guide https://www.python.org/dev/peps/pep-0008/#imports

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 3]
```diff
@@ -62,6 +46,167 @@
...
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```
Comment (trishnaguha) [inline bodhi/server/models.py]: Move the docstring before import statement https://www.python.org/dev/peps/pep-0008/#imports
Here is  duplication of docstring https://github.com/ankit01ojha/bodhi/blob/b7fdff2e6f66d8a487848f8424189bec798a55c9/bodhi/server/models.py#L49-L77

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 4]
```diff
@@ -61,6 +60,135 @@
 except ImportError:
     log.warning("Could not import 'rpm'")
 
+import re
```
Comment (trishnaguha) [inline bodhi/server/models.py]: Importing this twice?

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 5]
```diff
@@ -61,6 +60,135 @@
...
+
+from sqlalchemy.types import SchemaType, TypeDecorator, Enum
```
Comment (trishnaguha) [inline bodhi/server/models.py]: Could you move this to https://github.com/ankit01ojha/bodhi/blob/0ce799e298fa85d3f734debd4a5b439a50275a3a/bodhi/server/models.py#L45 ?

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py (hunk 5 above)
Comment (ankit01ojha) [reply bodhi/server/models.py]: Do I have to write this in place of https://github.com/ankit01ojha/bodhi/blob/0ce799e298fa85d3f734debd4a5b439a50275a3a/bodhi/server/models.py#L45 or before it?

File: bodhi/server/models.py [hunk 6]
```diff
@@ -14,6 +12,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```
Comment (trishnaguha) [inline bodhi/server/models.py:15]: There is still duplication here https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L173-L187

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 7]
```diff
@@ -1,5 +1,3 @@
-# encoding: utf-8
```
Comment (trishnaguha) [inline bodhi/server/models.py:1]: Do not remove `encoding: utf-8` :).

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Comment by trishnaguha** [comment]
Can you please fix the branch conflict? And squash and rebase your commits before pushing?

File: .gitignore [hunk 8]
```diff
@@ -21,3 +21,5 @@ dogpile-cache.dbm*
 html-docs
 masher/
 Vagrantfile
+.idea/
```
Comment (trishnaguha) [inline .gitignore]: It would be more perfect to have `*.idea/*` instead.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 9]
```diff
@@ -61,6 +61,133 @@
...
+        t = self.dialect_impl(bind.dialect)
+        if t.impl.__class__ is not self.__class__ and isinstance(t, SchemaType):
+            t.impl.create(bind=bind, checkfirst=checkfirst)
+
+    def drop(self, bind=None, checkfirst=False):
+        """Issue DROP ddl for this type, if applicable."""
+        super(DeclEnumType, self).drop(bind, checkfirst)
+        t = self.dialect_impl(bind.dialect)
+        if t.impl.__class__ is not self.__class__ and isinstance(t, SchemaType):
+t.impl.drop(bind=bind, checkfirst=checkfirst)
```
Comment (trishnaguha) [inline bodhi/server/models.py]: Wrong indentation here.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 10]
```diff
@@ -61,6 +61,133 @@
 except ImportError:
     log.warning("Could not import 'rpm'")
```
Comment (trishnaguha) [inline bodhi/server/models.py:62]: Don't we want to keep the first try block here? https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L28-L32

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py (hunk 10 above)
Comment (ankit01ojha) [reply bodhi/server/models.py:62]: Yes, we can.

**Comment by ankit01ojha** [comment]
Sorry it again went two commits :(

File: bodhi/server/models.py (hunk 10 above)
Comment (trishnaguha) [reply bodhi/server/models.py:62]: This looks pretty messed up now. This is not what I meant https://github.com/ankit01ojha/bodhi/blob/b416dcbd62073fa3a495dcd9975c096504e92a0b/bodhi/server/models.py#L53-L60

Comment (ankit01ojha) [reply bodhi/server/models.py:62]: Should I undo it ?

Comment (ankit01ojha) [reply bodhi/server/models.py:62]: If I put the first try block here then can you please tell me where this try block should go:https://github.com/ankit01ojha/bodhi/blob/8f4a951731fc8a2b32439092ae3587a404a3e076/bodhi/server/models.py#L59-L62   ?

Comment (trishnaguha) [reply bodhi/server/models.py:62]: We should avoid having `try-block` in the middle of the `import` statements.

**Comment by bowlofeggs** [comment]
@ankit01ojha This pull request has been sitting for a while. Do you still intend to work on it?

**Comment by ankit01ojha** [comment]
@bowlofeggs sorry, for keeping this PR still for a while, I was having issues with my laptop that's why I was not able to work. Now its fixed , so i'll work.

**Comment by ankit01ojha** [comment]
@bowlofeggs I have closed this PR as it looks pretty messed up :) 
A new PR has been opened https://github.com/fedora-infra/bodhi/pull/1190.

**Comment by bowlofeggs** [comment]
On Fri, 2016-12-23 at 21:03 -0800, Ankit Raj Ojha wrote:
> @bowlofeggs I have closed this PR as it looks pretty messed up :)
> A new PR has been opened #1190.

Hello!

For next time, you can alter history in your own branch and force push
it to update the pull request. This will prevent you from having to
open a new one when you want to make changes.

**Comment by ankit01ojha** [comment]
@bowlofeggs Thanks, I'll keep that for the next time. :)

**Pull request closed without merging**